src/cachemachine/types.py

```diff
@@ -96,6 +105,30 @@ def dump(self) -> List[Dict[str, str]]:
         return [asdict(i) for i in self]
 
 
+@dataclass
+class ImageEntry:
+    """Container for an entry in the cache from kubernetes.
+
+    Since we build this up over a loop, we might have one part first,
+    then the other part, so all of these are optional.  Call valid
+    when you think you are done to check that you have all the parts
+    you need.
+    """
+
+    """SHA256 image hash of the image entry."""
+    image_hash: Optional[str]
+
+    """List of tags the image is known by."""
+    tags: List[str]
+
+    def __init__(self) -> None:
+        self.image_hash = None
+        self.tags = []
```

Comment:
Nit: Dataclasses shouldn't have explicit constructors, but instead should set the defaults when the variable is defined. So instead:

```
from dataclasses import default_factory

# ...

image_hash: Optional[str] = None
tags: List[str] = default_factory(list)
```